packages/web/src/pages/pay-and-earn-page/PayAndEarnPage.tsx

```diff
@@ -6,6 +11,21 @@ import { PayAndEarnPageProps } from './types'
 
 export const PayAndEarnPage = (props: PayAndEarnPageProps) => {
   const isMobile = useIsMobile()
+  const dispatch = useDispatch()
+  const { recoveryStatus, refresh } = useUSDCBalance()
+
+  // Refresh balance on successful recovery
+  useEffect(() => {
+    if (recoveryStatus === Status.SUCCESS) {
+      refresh()
+    }
+  }, [recoveryStatus, refresh])
+
+  // Always check for recoverable USDC on page load
+  useEffect(() => {
+    dispatch(buyUSDCActions.startRecoveryIfNecessary())
```

Comment:
No polling? Can we do polling while the window is in focus/active tab?

Reply:
We _can_ do it, but I'm not sure it'll be super necessary. We already initiate recovery upon closing the modal and upon loading the page. The situations we would catch with polling are:
* Recovery failed even with retries
* Some money ended up in your root wallet from a source outside of the current browser session.

Putting that against the need to selectively disable the polling when the window isn't visible and while the withdrawal modal is open seemed like it might not be worth the potential issues it could cause.
But happy to give it a shot, though I might add it in a follow-up PR if that's okay with you.